certmanager/util.go

```diff
@@ -18,10 +18,40 @@ import (
 	"crypto"
 	"crypto/x509"
 	"encoding/pem"
+	"net/http"
 
 	"github.com/golang/glog"
 )
 
+const (
+	// The GCE metadata service URI to get identity token.
+	identityUri = "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity"
+)
+
+// Get the GCE VM identity jwt token from its metadata server.
+// Note: this function only works in a GCE VM environment.
+func GetIdentityToken(aud string) ([]byte, error) {
+	req, err := http.NewRequest("GET", identityUri+aud, nil)
+	if err != nil {
+		glog.Errorf("Fail to construct the http request: %s", err)
+		return nil, err
+	}
+	req.Header.Set("Metadata-Flavor", "Google")
```

Comment:
Can we utilize anything from https://github.com/GoogleCloudPlatform/google-cloud-go/blob/master/compute/metadata/metadata.go to avoid doing these low-level stuff?